Curriculum
============

**_Coming of age: ten years of next-generation sequencing technologies_**  
Goodwin _et. al._, Nature Reviews Genetics 2016

Figures 1-4 from
**_Sequencing technologies — the next generation_**  
Metzker, Nature Reviews Genetics 2009

**_An Extensive Evaluation of Read Trimming Effects on Illumina NGS Data Analysis_**  
Fabbro _et. al._, PLoS ONE 2013

**_Sequencing depth and coverage: key considerations in genomic analyses_**  
Sims _et. al._, Nature Reviews Genetics 2014

**_A field guide to whole‐genome sequencing, assembly and annotation_**  
Ekblom and Wolf, Evolutionary Applications 2014

**_The Theory and Practice of Genome Sequence Assembly_**  
Simpson and Pop, Annual Review of Genomics and Human Genetics 2015

**_Benchmarking short sequence mapping tools_**  
Hatem  _et. al._, BMC Bioinformatics 2013

**_Genotype and SNP calling from next-generation sequencing data_**  
Nielsen _et. al._, Nature Reviews Genetics 2011

**_Exome sequencing identifies the cause of a mendelian disorder_**  
Ng _et. al._, Nature Genetics 2009

**_A survey of best practices for RNA-seq data analysis_**  
Conesa _et. al._ Genome Biology 2016

**_The power and promise of RNA‐seq in ecology and evolution_**  
Todd _et. al._, Molecular Ecology 2016

**_The dilemma of choosing the ideal permutation strategy while estimating statistical significance of genome-wide enrichment_**  
Subhajyoti _et. al._, Briefings in Bioinformatics 2013

**_Transcriptionally active regions are the preferred targets for chromosomal HPV integration in cervical carcinogenesis_**  
Christiansen _et. al._, PLoS ONE 2015

**_GSuite HyperBrowser: integrative analysis of dataset collections across the genome and epigenome_**  
Simovski _et. al._, BioRxiv preprint

**_Ten Simple Rules for Reproducible Computational Research_**  
Sandve _et. al._, PLoS Computational Biology 2013

**_A Quick Guide to Organizing Computational Biology Projects_**  
Noble, PLoS Computational Biology 2009


